src/model/sql_query/row.rs

```diff
@@ -0,0 +1,304 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+// [Row] in sql query
+
+use std::collections::BTreeMap;
+
+use arrow::{
+    array::{
+        ArrayRef, BinaryArray, BooleanArray, Int16Array, Int32Array, Int64Array, Int8Array,
+        StringArray, Time32MillisecondArray, TimestampMillisecondArray, UInt16Array, UInt32Array,
+        UInt64Array, UInt8Array,
+    },
+    datatypes::{DataType, TimeUnit},
+    record_batch::RecordBatch,
+};
+use paste::paste;
+
+use crate::{model::value::Value, Error, Result};
+
+#[derive(Debug, PartialEq)]
+pub struct Row {
+    // It is better to iterate in a fixed order, also can save memory.
+    values: BTreeMap<String, Value>,
```

Comment:
The original order of the query result may be missing if we use `BTreeMap` rather than `Vec`.